install/kubernetes/cilium/templates/cilium-nodeinit-daemonset.yaml

```diff
@@ -242,8 +242,25 @@ spec:
                   docker kill "${container_id}" || true
                 done
               else
-                # COS-beta (with containerd). Some versions of COS have crictl in /home/kubernetes/bin.
-                for f in `find /var/lib/cni/networks/ -type f ! -name lock ! -name last_reserved_ip.0`; do PATH="${PATH}:/home/kubernetes/bin" crictl stopp "$(sed 's/\r//;1q' $f)" || true; done
+                if [ -d /var/lib/cni/networks/ ]; then
```

Comment:
I don't think any of these `-d` conditions are necessary. Just run all `find`s unconditionally with `2>/dev/null || true`. That way, if some host has multiple of these directories, but only uses one of them, this won't break.

Reply:
Will change it to:
`              else
                # COS-beta (with containerd). Some versions of COS have crictl in /home/kubernetes/bin.
                echo "Checking /var/lib/cni/networks/"
                for f in `find /var/lib/cni/networks/ -type f ! -name lock ! -name last_reserved_ip.0 2>/dev/null || true`; do PATH="${PATH}:/home/kubernetes/bin" crictl stopp "$(sed 's/\r//;1q' $f)" || true; done
                # AKS 1.18.X uses /var/lib/cni/cache/results and docker
                echo "Checking /var/lib/cni/cache/results/"
                for f in `find /var/lib/cni/cache/results/ -type f ! -name "cilium-*" ! -name "cni-loopback-*" -printf "%f\n" 2>/dev/null || true`; do IFS="-" read -a id <<<"$f" ; docker kill ${id[1]} || true; done
                # AKS 1.19.X and higher uses /var/lib/cni/results. Tested with: AKS 1.19.X, 1.20.X, 1.21.X
                echo "Checking /var/lib/cni/results/"
                for f in `find /var/lib/cni/results/ -type f ! -name "cilium-*" ! -name "cni-loopback-*" 2>/dev/null || true`; do PATH="${PATH}:/home/kubernetes/bin" crictl stopp "$(sed 's/\r//;1q' $f | jq .containerId | tr -d '"')" || true; done
              fi`